cmd/configure_raid.go

```diff
@@ -13,31 +14,52 @@ var configureRaidCmd = &cobra.Command{
 	Short: "Configures various types of RAID",
 	Long:  "Configures various types of RAID",
 	Run: func(cmd *cobra.Command, args []string) {
-		ctx := context.Background()
+		ctx := command.NewContextWithLogger(logger)
 
-		blockDeviceFiles := GetStringSlice(cmd, "devices")
+		if GetBool(cmd, "delete") {
+			raidArray := model.RaidArray{
+				Name: GetString(cmd, "name"),
+			}
 
-		blockDevices, err := model.NewBlockDevices(blockDeviceFiles...)
-		if err != nil {
-			logger.Fatalw("Failed to GetBlockDevices", "err", err, "devices", blockDeviceFiles)
-		}
+			if out, err := raidArray.Delete(ctx, GetString(cmd, "raid-type")); err != nil {
+				logger.Fatalw("failed to create raid array", "err", err, "array", raidArray, "output", out)
+			}
+		} else {
+			var blockDeviceIDs []int
 
-		raidArray := model.RaidArray{
-			Name:    GetString(cmd, "name"),
-			Devices: blockDevices,
-			Level:   GetString(cmd, "raid-level"),
-		}
+			for _, d := range GetStringSlice(cmd, "devices") {
+				intBlockDevice, err := strconv.Atoi(d)
+				if err != nil {
+					logger.Fatalw("failed to convert device id string to int", "err", err, "blockDeviceID", d)
+				}
+
+				blockDeviceIDs = append(blockDeviceIDs, intBlockDevice)
+			}
+
+			// TODO(splaspood) Handle looking up devices using ironlib/mvcli to generate this list?
+			blockDevices, err := model.NewBlockDevicesFromPhysicalDeviceIDs(blockDeviceIDs...)
+			if err != nil {
+				logger.Fatalw("failed to gather block devices from physical ids", "err", err, "devices", blockDeviceIDs)
+			}
+
+			raidArray := model.RaidArray{
+				Name:    GetString(cmd, "name"),
+				Devices: blockDevices,
+				Level:   GetString(cmd, "raid-level"),
+			}
 
-		if err = raidArray.Create(ctx, GetString(cmd, "raid-type")); err != nil {
-			logger.Fatalw("failed to create raid array", "err", err, "array", raidArray)
+			if err = raidArray.Create(ctx, GetString(cmd, "raid-type")); err != nil {
```

Comment:
It would just be create or delete, presumably...   I went back and forth on this a bunch of times...   I felt if it was going to be >= 3 'actions' I'd go for an additional level of subcommands, but since it's just two a bool flag felt better...   we are applying a raid 'config' and maybe that config is a deletion rather than a creation.